src/NamingStrategy/NamingStrategyInterface.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Plugin\Vcr\NamingStrategy;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * In charge of giving a deterministic name to a request.
```

Comment:
the purpose of that name is a filename, right? can we be more explicit here? "Provide a deterministic and unique filename for a request"?

Reply:
What about `Provide a deterministic and unique filename / database ID for a request`? Some might create a recorder for database too.

Reply:
then lets say "Provide a deterministic and unique identifier for a request. The identifier must be safe to use with a filesystem."